### PR TITLE
Automatically create backport PR if oauth token is set

### PR DESCRIPTION
**Pull request by Mariatta**
- Add `gidgethub` and `requests` as dependencies.

Closes https://github.com/python/core-workflow/issues/100

### CODE REVIEW

**Comment by Mariatta** [comment]
https://github.com/python/cpython/pull/3224 was created via this method, using my own personal access token.

**Comment by terryjreedy** [comment]
I have read the readme, but it is not clear to me what 'create automatically' means -- what will happen, when and where -- and what a token on my machine has to do with it.

**Comment by Mariatta** [comment]
Thanks @terryjreedy. The idea is instead of opening a web browser where we have to click the 'Create pull request button', cherry picker will create the PR. So no web browser will be opened at all.
cherry_picker will print out the URL of the newly created pull request.

The token is needed by GitHub API as a way to authenticate your GitHub account.

**Comment by terryjreedy** [comment]
Thanks for explaining.  Since I like to have the PR open in my browser, and clicking the button is an easy way to get that, I will probably pass.

**Comment by Mariatta** [comment]
No prob :) It's backward compatible, so the previous way still works as is.

Potentially we can use this for the cherry-picking bot, where the backport will happen automatically after we merge the PR on `master`. (Issue https://github.com/python/core-workflow/issues/8)

File: cherry_picker/readme.rst [hunk 1]
```diff
@@ -53,6 +57,19 @@ repository are pushed to ``origin``. If this is incorrect, then the correct
 remote will need be specified using the ``--pr-remote`` option (e.g.
 ``--pr-remote pr`` to use a remote named ``pr``).
 
+Personal Access Token (optional)
+--------------------------------
+
+``cherry_picker`` can automatically create the PR, if the ``GH_AUTH`` token is set
```
Comment (brettcannon) [inline cherry_picker/readme.rst]: Is this considered safe and best practice for storing an auth token like this? I honestly don't know, which is why I'm asking. 😄

Comment (Mariatta) [reply cherry_picker/readme.rst]: I don't really know 😅 
This is really so I can use it for the cherry picker bot later.
Maybe I should keep this as hidden and secret functionality, and not mention it at all in the readme?

Comment (brettcannon) [reply cherry_picker/readme.rst]: Yeah, either don't document it or be clear that one should be careful about how that environment variable is stored.

Comment (Mariatta) [reply cherry_picker/readme.rst]: Ok I'm going with "not documenting it" since it's easier to do than coming up with warning messages 😅

**Comment by Mariatta** [comment]
I didn't expect the Spanish Inquisition!

File: cherry_picker/cherry_picker/cherry_picker.py [hunk 2]
```diff
@@ -6,9 +6,13 @@
 import subprocess
 import webbrowser
 import sys
+import requests
+
+from gidgethub.sansio import create_headers
```
Comment (brettcannon) [inline cherry_picker/cherry_picker/cherry_picker.py]: Not important, but you shifted from importing modules to importing objects with this import line.

File: cherry_picker/cherry_picker/cherry_picker.py [hunk 3]
```diff
@@ -150,7 +155,27 @@ def push_to_remote(self, base_branch, head_branch):
         except subprocess.CalledProcessError:
             click.echo(f"Failed to push to {self.pr_remote} \u2639")
         else:
-            self.open_pr(self.get_pr_url(base_branch, head_branch))
+            if os.getenv("GH_AUTH"):
+                self.create_gh_pr(base_branch, head_branch, commit_message)
+            else:
+                self.open_pr(self.get_pr_url(base_branch, head_branch))
+
+    def create_gh_pr(self, base_branch, head_branch, commit_message):
```
Comment (brettcannon) [inline cherry_picker/cherry_picker/cherry_picker.py]: If you're already checking for the envvar above, would it make sense to make it a required argument and just pass it in explicitly?

Comment (Mariatta) [reply cherry_picker/cherry_picker/cherry_picker.py]: Ok, I made the token a required argument.

File: cherry_picker/cherry_picker/cherry_picker.py [hunk 4]
```diff
@@ -8,7 +8,7 @@
 import sys
 import requests
 
-from gidgethub.sansio import create_headers
+import gidgethub
```
Comment (brettcannon) [inline cherry_picker/cherry_picker/cherry_picker.py]: This should either be `import gidgethub.sansio` or `from gidgethub import sansio`; the fact that the package has an import is undocumented.

File: cherry_picker/cherry_picker/cherry_picker.py [hunk 5]
```diff
@@ -155,13 +155,19 @@ def push_to_remote(self, base_branch, head_branch, commit_message=""):
...
+            gh_auth = os.getenv("GH_AUTH")
+            if gh_auth:
+                self.create_gh_pr(base_branch, head_branch, commit_message,
+                                  gh_auth)
             else:
                 self.open_pr(self.get_pr_url(base_branch, head_branch))
 
-    def create_gh_pr(self, base_branch, head_branch, commit_message):
-        request_headers = create_headers(self.username, oauth_token=os.getenv("GH_AUTH"))
+    def create_gh_pr(self, base_branch, head_branch, commit_message, gh_auth):
```
Comment (brettcannon) [inline cherry_picker/cherry_picker/cherry_picker.py]: Is there any worry the order of these arguments will be non-obvious? I.e. should any of them be keyword-only arguments?

Comment (Mariatta) [reply cherry_picker/cherry_picker/cherry_picker.py]: Hmm, ok I made commit_message and gh_auth as keyword only arguments.

File: cherry_picker/cherry_picker/cherry_picker.py (hunk 4 above)
Comment (Mariatta) [reply cherry_picker/cherry_picker/cherry_picker.py]: Changed to`from gidgethub import sansio`

**Pull request merged**